spring-cloud-commons/src/main/java/org/springframework/cloud/configuration/SSLContextFactory.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.configuration;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.UnrecoverableKeyException;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.http.ssl.SSLContextBuilder;
+
+import org.springframework.core.io.Resource;
+
+public class SSLContextFactory {
```

Comment:
@spencergibb what do you think about adding some autocofiguration for this class and creating a bean for it?

Reply:
Not sure. How would it be activated? Many projects depend on commons, but this will only be slated to be in config and netflix for now.

Reply:
The bean would be created and then optionally consumed by projects that need it.

Reply:
Is rather not create it when most consuming projects won't use it.